src/pybind/mgr/mgr_module.py

```diff
@@ -465,7 +400,92 @@ def register(self, instance=False):
         )
 
 
-class MgrStandbyModule(ceph_module.BaseMgrStandbyModule):
+class CPlusPlusHandler(logging.Handler):
+    def __init__(self, module_inst):
+        super(CPlusPlusHandler, self).__init__()
+        self._module = module_inst
+        self.setFormatter(logging.Formatter("[{}] [%(levelname)-4s] [%(name)s] %(message)s"
+                          .format(module_inst.module_name)))
+
+    def emit(self, record):
+        if record.levelno >= self.level:
+            self._module._ceph_log(self.format(record))
+
+
+class FileHandler(logging.FileHandler):
```

Comment:
Have you checked how this works with the default log rotation? (https://github.com/ceph/ceph/blob/master/src/logrotate.conf)

In the past I had lots of issues with Python loggers and logrotate, missing traces during rotation, etc.

Reply:
Not really, I guess if there are issues, they will still exist with this implementation.